opensm/osm_perfmgr.c

```diff
@@ -705,6 +705,12 @@ static void perfmgr_query_counters(cl_map_item_t * p_map_item, void *context)
 
 	OSM_LOG_ENTER(pm->log);
 
+    if (osm_exit_flag) {
+		OSM_LOG(pm->log, OSM_LOG_INFO,
+			"OpenSM is exiting!\n");
+        goto Exit;
```

Comment:
Exit releases lock that was not previously acquired.

Reply:
I have not seen this performance manager issue in another place for now.
I moved the code after acquiring the clock.

Reply:
There's no need to acquire the lock just to give it up to reuse the Exit. Just add Exit2 label before the OSM_LOG_EXIT and goto Exit2 here.

It looks to me that same issue also exists in the other PerfMgt routines which send MADs and wait with no timeout but we can deal with that later if need be.